proarc-common/src/test/java/cz/cas/lib/proarc/common/export/sip/NdkSipExportTest.java

```diff
@@ -0,0 +1,193 @@
+                                } catch (IOException e) {
+                                    throw new RuntimeException(e);
+                                }
+                            }
+
+                            @Override
+                            public <T> T getEntity(Class<T> c) {
+                                return null;
+                            }
+
+                            @Override
+                            public String getType() {
+                                return null;
+                            }
+
+                            @Override
+                            public void close() {
+
+                            }
+                        };
+                    }
+                };
+            }
+        };
+
+        MetaModelRepository.setInstance("ndk", "ndkEborn");
+    }
+
+    @Test
+    public void export() throws Exception {
+        NdkExport export = new NdkSipExport(remoteStorage, appConfig.getNdkExportOptions());
+
+        String pid = "uuid:acd66301-4e75-4d12-9d98-b323ff5beee9";
+
+        assertTrue("Junit didn't create a temporary folder", folder.getRoot().exists());
+
+        List<NdkExport.Result> resultsList = export.export(folder.getRoot(), Collections.singletonList(pid),
+                true, true, null);
+
+        for (NdkExport.Result result : resultsList) {
+            if (result.getValidationError() != null) {
+                throw result.getValidationError();
+            }
+        }
+        resultsList.get(0).getTargetFolder();
+
+        String packageId = "123";
+        Path sip = folder.getRoot().toPath().resolve(StringUtils.removeStart(pid, "uuid:")).resolve(packageId);
+        validatePackage(sip);
+    }
+
+    private void validatePackage(Path sip) throws Exception {
+        assertTrue("No SIP package", Files.isDirectory(sip));
+
+        String identifier = sip.getFileName().toString();
+
+        assertTrue("No original files", Files.list(sip.resolve("original")).count() > 0);
+        assertTrue("No metadata files", Files.list(sip.resolve("metadata")).count() > 0);
+        assertTrue("No info.xml", Files.exists(sip.resolve("info_" + identifier + ".xml")));
+        assertTrue("No pdf file", Files.exists(sip.resolve("original/oc_" + identifier + ".pdf")));
+        assertTrue("No mods file", Files.exists(sip.resolve("metadata/mods_volume.xml")));
+
+        List<String> errors = MetsUtils.validateAgainstXSD(sip.resolve("info_" + identifier + ".xml").toFile(), Info.class.getResourceAsStream("info.xsd"));
+        assertTrue(errors.toString(), errors.isEmpty());
+
+        JAXBContext jContext = JAXBContext.newInstance(Info.class);
+        Unmarshaller unmarshallerObj = jContext.createUnmarshaller();
+        Info info = (Info) unmarshallerObj.unmarshal(sip.resolve("info_" + identifier + ".xml").toFile());
+        assertTrue(info.getMetadataversion() >= 2.2);
```

Comment:
Asi to nedělá problém, ale porovnává se double a float, asi by bylo lepší použít 2.2f.